src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/CsoAccountServiceImpl.java

```diff
@@ -17,40 +18,87 @@
 
     private final AccountFacadeBean accountFacadeBean;
     private final RoleRegistryPortType roleRegistryPortType;
+    private final BCeIDServiceSoap bCeIDServiceSoap;
     private final AccountDetailsMapper accountDetailsMapper;
 
-    public CsoAccountServiceImpl(AccountFacadeBean accountFacadeBean, RoleRegistryPortType roleRegistryPortType,
+    public CsoAccountServiceImpl(AccountFacadeBean accountFacadeBean,
+                                 RoleRegistryPortType roleRegistryPortType,
+                                 BCeIDServiceSoap bCeIDServiceSoap,
                                  AccountDetailsMapper accountDetailsMapper) {
 
         this.accountFacadeBean = accountFacadeBean;
         this.roleRegistryPortType = roleRegistryPortType;
+        this.bCeIDServiceSoap = bCeIDServiceSoap;
         this.accountDetailsMapper = accountDetailsMapper;
     }
 
     @Override
-    public AccountDetails getAccountDetails(String userGuid) throws NestedEjbException_Exception {
+    public AccountDetails getAccountDetails(String userGuid, String bceidAccountType) throws NestedEjbException_Exception {
 
         if (StringUtils.isEmpty(userGuid)) return null;
 
+        AccountDetails accountDetails = getCsoDetails(userGuid);
+        if (null == accountDetails) {
+            accountDetails = getBCeidDetails(userGuid, bceidAccountType);
+        }
+
+        return accountDetails;
+    }
+
+    private AccountDetails getCsoDetails(String userGuid) throws NestedEjbException_Exception  {
+
         AccountDetails accountDetails = null;
+
         List<ClientProfile> profiles = accountFacadeBean.findProfiles(userGuid);
-        //An account must only one profile associated to proceed
+        // An account must only one profile associated to proceed
         if (profiles.size() == 1) {
-            accountDetails = accountDetailsMapper.toCsoAccountDetails(profiles.get(0), hasFileRole(userGuid));
+            accountDetails = accountDetailsMapper.toAccountDetails(profiles.get(0), hasFileRole(userGuid));
         }
         else if (profiles.size() > 1) {
             throw new CSOHasMultipleAccountException(profiles.get(0).getClientId().toString());
         }
 
         return accountDetails;
+    }
 
+    private AccountDetails getBCeidDetails(String userGuid, String accountType) {
+
+        AccountDetails accountDetails = null;
+        BCeIDAccountTypeCode accountTypeCode = getBCeIDAccountType(accountType);
+
+        if (accountTypeCode != BCeIDAccountTypeCode.VOID) {
+            AccountDetailRequest request = new AccountDetailRequest();
+            request.setOnlineServiceId("62B2-5550-4376-4DA7");
```

Comment:
so this supposed to be a hard coded value?

Reply:
Yeah, we might get our own service value, but we might just use CSO's for now.